Drivers/Src/MotorController.c

```diff
@@ -31,36 +31,26 @@ void MotorController_Drive(uint32_t newVelocity, uint32_t motorCurrent){
     BSP_CAN_Write(CAN_2, MOTOR_DRIVE, data, 8);
 }
 
-bool MotorController_Read(CANbuff *message){
+error_t MotorController_Read(CANbuff *message){
     uint32_t id;
     uint8_t data[8] = {0};
     uint32_t length = BSP_CAN_Read(CAN_2, &id, data);
     uint32_t firstSum = 0;
     uint32_t secondSum = 0;
     if(length>0){
         message->id = id;
-        if(length < 5){
-            for(int i = 0; i < length; i++){
-                firstSum += data[i] & 0xFF;
-                if(i != length-1){
-                    firstSum = firstSum << 8;
-                }
+        //get first number (bits 0-31)
+        for(int j = 0; j < 4; j++){
+            firstSum += data[j] & 0xFF;
+            if(j != 3){
```

Comment:
Some magic numbers floating around here. Try to define these values. Something like defining the 8 (size of data array) as a MAX_CAN_LEN or smth and then making the other values relative to that. Easier to read and change if necessary.